app/assets/javascripts/components/articles/articles_handler.jsx

```diff
@@ -19,10 +20,21 @@ const ArticlesHandler = createReactClass({
   },
 
   componentWillMount() {
-    ServerActions.fetch('articles', this.props.course_id);
+    state.limit = 500;
+    state.article_count = 501; // Make sure Show More button is shown at least once
+    // In case the following fails to resolve before rendering
+    $.getJSON(`/courses/${this.props.course_id}/article_count.json`, (data) => {
```

Comment:
I don't like the idea of adding an extra query for every page load, just to fetch the limit which will only be passed a small portion of the time. I think a better strategy is the one we use for revisions (implemented in the revisions reducer) of comparing the actual number of articles received with the limit, and assuming there are more articles only if the actual number reaches the limit.

Reply:
@ragesoss but articles are present within article_list.js I don't suppose I have the this.props.articles on hand in here?